include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -401,7 +401,10 @@ class transform_iterator
 {
   private:
     _Iter __my_it_;
-    const _UnaryFunc __my_unary_func_;
+    _UnaryFunc __my_unary_func_;
+
+    static_assert(std::is_invocable_v<const _UnaryFunc, typename std::iterator_traits<_Iter>::reference>,
```

Comment:
We don't support `std::invoke` semantics (though, we could) for `operator*`. So, this `static_assert` may return you `true` for wider amount of scenarios that we actually support. I am not sure if it's a problem or not.